bin/run_docker_test

```diff
@@ -60,63 +76,102 @@ def main():
-            raise RunDockerTestError(
-                "Aborting.. Could not inspect test services '{}'. Do "
-                "`docker ps -a` to list residual containers.".format(
-                    args.test_service))
-
-        end_time = time.time()
-        delta_time = end_time - start_time
-        start_time = end_time
-        timeout = int(timeout - delta_time)
-
-        # Clean up
-        try:
-            print("Running: {}".format(compose_down))
-            run(compose_down, check=True, timeout=timeout)
-        except CalledProcessError:
-            raise RunDockerTestError(
-                "Failed to remove all docker containers. Do `docker ps -a` to "
-                "list residual containers.")
-
-        exit(int(success))
+        if not args.clean:
+            exit_status = 0
+            timer.start()
+
+            LOGGER.info("Bringing up with %s", str(compose_up))
+            try:
+                # 1. Run the tests
+                subprocess.run(compose_up, check=True, timeout=timer.remaining())
+
+            except FileNotFoundError as err:
+                LOGGER.error("Bad docker-compose up command")
+                LOGGER.exception(err)
+                exit(1)
+
+            except subprocess.CalledProcessError as err:
+                LOGGER.error("Failed to start test.")
+                LOGGER.exception(err)
+                exit_status = 1
+
+            except subprocess.TimeoutExpired as err:
+                LOGGER.error("Test timed out.")
+                LOGGER.exception(err)
+                exit_status = 1
+
+            if exit_status == 0:
+                LOGGER.info("Getting result with: %s", str(inspect))
+                try:
+                    # 2. Get the exit code of the test container
+                    exit_status = subprocess.run(
+                        inspect, stdout=subprocess.PIPE,
+                        timeout=timer.remaining(), check=True
+                    ).stdout.decode().strip()
+
+                except FileNotFoundError as err:
+                    LOGGER.error("Bad docker inspect command")
+                    LOGGER.exception(err)
+                    exit_status = 1
+
+                except subprocess.CalledProcessError as err:
+                    LOGGER.error("Failed to retrieve exit status of test.")
+                    LOGGER.exception(err)
+                    exit_status = 1
+
+                except subprocess.TimeoutExpired as err:
+                    LOGGER.error("Retrieving exit status timed out.")
+                    LOGGER.exception(err)
+                    exit_status = 1
+
+        else: # cleaning
+            exit_status = 0
+
+        LOGGER.info("Shutting down with: %s", str(compose_down))
+        shutdown_success = False
+        for i in range(2):
+            if shutdown_success == False:
+
+                # Always give compose down at least 60 seconds to cleanup
+                timeout = max(timer.remaining(), 60)
```

Comment:
Sure, sorry I was pushing/pulling to my Intel machine where I had PoET in a bad state to verify.